internal/attestation/aws/snp/validator_test.go

```diff
@@ -7,41 +7,71 @@ SPDX-License-Identifier: AGPL-3.0-only
 package snp
 
 import (
+	"bytes"
 	"context"
+	"crypto"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/hex"
 	"encoding/json"
+	"encoding/pem"
 	"errors"
+	"fmt"
+	"regexp"
 	"testing"
 
-	"github.com/aws/aws-sdk-go-v2/feature/ec2/imds"
-	"github.com/aws/aws-sdk-go-v2/service/ec2"
-	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/edgelesssys/constellation/v2/internal/attestation"
+	"github.com/edgelesssys/constellation/v2/internal/attestation/aws/snp/testdata"
+	"github.com/edgelesssys/constellation/v2/internal/attestation/snp"
 	"github.com/edgelesssys/constellation/v2/internal/attestation/vtpm"
+	"github.com/edgelesssys/constellation/v2/internal/config"
+	"github.com/edgelesssys/constellation/v2/internal/logger"
+	"github.com/google/go-sev-guest/abi"
+	"github.com/google/go-sev-guest/proto/sevsnp"
+	spb "github.com/google/go-sev-guest/proto/sevsnp"
+	"github.com/google/go-sev-guest/verify"
 	"github.com/google/go-tpm-tools/proto/attest"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
-func TestGeTrustedKey(t *testing.T) {
+const (
+	// validAKDigest is the AK hash of the public key embedded in 'reportValid'.
+	validAKDigest = "032635613c8e331fa29e096371910fe6a1f69383dda02c9461400a70b66d87a3da5dd863002522be43afc34f2c233989bd6e401e351d10d7cc800d6f5dfcf019"
+)
+
+func TestGetTrustedKey(t *testing.T) {
+	validator := &Validator{ask: nil, ark: nil, reportValidator: stubawsValidator{}}
```

Comment:
```suggestion
	validator := func() *Validator {return &Validator{ask: nil, ark: nil, reportValidator: stubawsValidator{}}}
```
nit: prevent issues arising from shared state